### PR TITLE
fix: re-enable parquet row group pruning

### PR DESCRIPTION
**Pull request by alamb**
Follow on to https://github.com/influxdata/influxdb_iox/pull/2838

Resolves https://github.com/influxdata/influxdb_iox/issues/2163 now that https://github.com/apache/arrow-rs/issues/641 has been fixed.

Also improves use of DataFusion's new file source API (Thanks @rdettai)

### CODE REVIEW

File: parquet_file/src/storage.rs [hunk 1]
```diff
@@ -276,34 +289,43 @@ impl Storage {
...
-            debug!(?predicate, "Skipping predicate pushdown due to XXX");
-        }
-        let predicate = None;
-
         let object_store = Arc::new(LocalFileSystem {});
+        let filters: Vec<_> = predicate.into_iter().collect();
+
+        // Read statistics from the parquet file so we can use parquet
+        // row group pruning. It might be cool to use the statistics
+        // stored in the IOx catalog to do this (evetually)
```
Comment (alamb) [inline parquet_file/src/storage.rs:297]:
```suggestion
        // stored in the IOx catalog to do this (eventually)
```

File: parquet_file/src/storage.rs [hunk 2]
```diff
@@ -276,34 +289,43 @@ impl Storage {
...
+        let format = ParquetFormat::default();
```
Comment (alamb) [inline parquet_file/src/storage.rs:298]: This uses the cool new `FileFormat` trait / API that was added to DataFusion to create the `ParquetExec` without having to do so directly. 👍

File: parquet_file/src/storage.rs [hunk 3]
```diff
@@ -276,34 +289,43 @@ impl Storage {
...
-        // https://github.com/influxdata/influxdb_iox/issues/2163
-        if predicate.is_some() {
-            debug!(?predicate, "Skipping predicate pushdown due to XXX");
-        }
-        let predicate = None;
-
         let object_store = Arc::new(LocalFileSystem {});
+        let filters: Vec<_> = predicate.into_iter().collect();
+
+        // Read statistics from the parquet file so we can use parquet
```
Comment (alamb) [inline parquet_file/src/storage.rs:295]: Previously if filters were passed down to DataFusion several of the `query_tests` tests would fail

**Comment by tustvold** [comment]
@alamb Is there a reason this is just sitting here orphaned?

**Comment by alamb** [comment]
> @alamb Is there a reason this is just sitting here orphaned?

@tustvold  yes it is because I haven't gotten around to writing a test as suggested by @NGA-TRAN  -- given I don't really expect this to have a major performance difference I haven't made writing one a priority

**Comment by alamb** [comment]
@tustvold  has an alternate plan

**Pull request closed without merging**